server/src/main/java/org/opensearch/rest/pagination/PaginatedQueryResponse.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.pagination;
+
+/**
+ * Pagination response metadata for a paginated query.
+ * @opensearch.internal
+ */
+public class PaginatedQueryResponse {
+
+    public static final String PAGINATED_RESPONSE_NEXT_TOKEN_KEY = "next_token";
+
+    /**
+     * String denoting the next_token of paginated response, which will be used to fetch next page (if any).
+     */
+    private final String nextToken;
+
+    /**
+     * String denoting the element which is being paginated (for e.g. shards, indices..).
+     */
+    private final String paginatedElement;
```

Comment:
make it enum, instead of Element, call it Entity

Reply:
Renamed in the revision, but didn't ENUMERATE it. I'm thinking this is very specific to the strategy/API which is getting paginated as to what it wants to display as part of response. So, thought to keep it as a string inside the strategy itself.

If you feel it's better to define ENUM for this, then I can update again.